Ginger/Ginger/ApplicationModelsLib/APIModels/APIModelBodyNodeSyncPage.xaml.cs

```diff
@@ -0,0 +1,301 @@
+                regexStringBuilder.Append(splitedSearchText[i] + "\\s*\\n*\\[\\s*\\n*");
+
+            splitedSearchText = regexStringBuilder.ToString().Substring(0, regexStringBuilder.ToString().Length-14).Split(']');
+            regexStringBuilder.Clear();
+            for (i = 0; i < splitedSearchText.Length; i++)
+                regexStringBuilder.Append(splitedSearchText[i] + "\\s*\\n*\\]\\s*\\n*");
+
+            //string regexFinalString = regexStringBuilder.ToString().Substring(0, regexStringBuilder.ToString().Length - 7) + ',';
+            string regexFinalString = regexStringBuilder.ToString().Substring(0, regexStringBuilder.ToString().Length - 14);
+            var regex = new Regex(regexFinalString);
+            Match match = regex.Match(mApplicationAPIModel.RequestBody);
+
+            if (match.Success)
+                nodeToDelete.stringNodeRange = new Tuple<int, int>(match.Index, match.Index + match.Length);
+            else
+                NodesToDeleteList.Remove(nodeToDelete);
+        }
+
+
+        private void DeleteOnlyParamsButton_Click(object sender, RoutedEventArgs e)
+        {
+            if ((bool)xRemoveAssociatedParams.IsChecked)
+                AddAssociatedParamsForDeletion();
+
+            DeleteParams();
+            _pageGenericWin.Close();
+        }
+
+        private void DeleteParamsAndBodyNodesButton_Click(object sender, RoutedEventArgs e)
+        {
+            if ((bool)xRemoveAssociatedParams.IsChecked)
+                AddAssociatedParamsForDeletion();
+
+            foreach (NodeToDelete xmlNode in NodesToDeleteList)
+            {
+                if(xmlNode.stringNodeRange.Item1 - RemovedCharsFromRequestBodyCounter > 0 )
+                    mApplicationAPIModel.RequestBody = mApplicationAPIModel.RequestBody.Remove(xmlNode.stringNodeRange.Item1 - RemovedCharsFromRequestBodyCounter, xmlNode.stringNodeRange.Item2 - xmlNode.stringNodeRange.Item1);
+                else
+                    mApplicationAPIModel.RequestBody = mApplicationAPIModel.RequestBody.Remove(0, xmlNode.stringNodeRange.Item2 - xmlNode.stringNodeRange.Item1);
+
+                RemovedCharsFromRequestBodyCounter += xmlNode.stringNodeRange.Item2 - xmlNode.stringNodeRange.Item1;
+            }
+            DeleteParams();
+
+            _pageGenericWin.Close();
+        }
+
+        private void DeleteParams()
+        {
+            foreach (AppModelParameter param in mParamsPendingDelete)
+                mApplicationAPIModel.AppModelParameters.Remove(param);
+        }
+
+        private void AddAssociatedParamsForDeletion()
+        {
+            foreach (NodeToDelete xmlNode in NodesToDeleteList)
+            {
+                Regex regex = null;
+                if (requestBodyType == ApplicationAPIUtils.eContentType.XML)
+                    regex = new Regex(@"{(.*?)\}");
+                else if(requestBodyType == ApplicationAPIUtils.eContentType.JSon)
+                    regex = new Regex(@"<(.*?)\>");
+
```

Comment:
you can't relay that the place holders will have {} or <>, this is only what we set as defualt but those are place holders and user can put what ever he wants

Reply:
Fixed